includes/Products/Sync.php

```diff
@@ -58,6 +57,66 @@ public function add_hooks() {
 	}
 
 
+	/**
+	 * Adds all eligible product IDs to the requests array to be created or updated.
+	 *
+	 * Uses the same logic that the feed handler uses to get a list of product IDs to sync.
+	 *
+	 * TODO: consolidate the logic to decide whether a product should be synced in one or a couple of helper methods - right now we have slightly different versions of the same code in different places {WV 2020-05-25}
+	 *
+	 * @see \WC_Facebook_Product_Feed::get_product_ids()
+	 * @see \WC_Facebook_Product_Feed::write_product_feed_file()
+	 *
+	 * @since 2.0.0-dev.1
+	 */
+	public function create_or_update_all_products() {
+
+		$product_ids        = [];
+		$parent_product_ids = [];
+
+		// loop through all published products and product variations to get their IDs
+		$args = [
+			'fields'         => 'id=>parent',
+			'post_status'    => 'publish',
+			'post_type'      => [ 'product', 'product_variation' ],
+			'posts_per_page' => -1,
+		];
+
+		foreach ( get_posts( $args ) as $post_id => $post_parent ) {
+
+			$product_ids[] = $post_id;
+
+			if ( 'product_variation' === get_post_type( $post_id ) ) {
+				$parent_product_ids[] = $post_parent;
+			}
+		}
+
+		// remove parent products because those can't be represented as Product Items
+		$product_ids = array_diff( $product_ids, $parent_product_ids );
+
+		// make sure the product should be synced and add it to the sync queue
+		foreach ( $product_ids as $product_id ) {
```

Comment:
This should also account for virtual products and variations.

Reply:
Added 3e42528281ecc50e9be67eb8b57aecbdfc21004c to account for virtual, confirmed no virtual products or variations are sent.